src/java/map/ConcurrentFloatMask.java

```diff
@@ -33,43 +35,71 @@ public ConcurrentFloatMask(ConcurrentFloatMask mask, long seed, String name) {
 	}
 
 	public ConcurrentFloatMask init(ConcurrentBinaryMask other, float low, float high) {
+		if (VisualDebugger.ENABLED) {
+			this.floatMask.init(other.getBinaryMask(), low, high);
+			return this;
+		}
 		return Pipeline.add(this, Arrays.asList(this, other), res -> {
 				return this.floatMask.init(((ConcurrentBinaryMask) res.get(1)).getBinaryMask(), low, high);
 			}
 		);
 	}
 
 	public ConcurrentFloatMask add(ConcurrentFloatMask other) {
+		if (VisualDebugger.ENABLED) {
+			this.floatMask.add(other.getFloatMask());
+			return this;
+		}
 		return Pipeline.add(this, Arrays.asList(this, other), res ->
 				this.floatMask.add(((ConcurrentFloatMask)res.get(1)).getFloatMask())
 		);
 	}
 
 	public ConcurrentFloatMask max(ConcurrentBinaryMask other) {
+		if (VisualDebugger.ENABLED) {
+			throw new UnsupportedOperationException();
+		}
 		return Pipeline.add(this, Arrays.asList(this, other), res ->
+				// Probably a bug, res should be BinaryMask
```

Comment:
probably correct

Reply:
can you just push the fix on master or smthn, might not want to mix it with this PR

Reply:
ok i removed the method because i makes no sense